### PR TITLE
Filter type

### LINKED ISSUE #999: Allow top-level filtering for MIRI MRS
It would be extremely useful to be able to select only programs that use the MIRI MRS as these program have unique issues compared to the MIRI imager.  At the top level 'Archived MIRI Images' page, it would therefore be useful to be able to downselect to show only programs that contain some MRS data instead of displaying all programs and having to click through each one to see which contain MRS vs imaging-only data.

### LINKED ISSUE #1028: Filter by exp_type
It might be nice to give the option to filter images based on exp_type. As I'm working on testing preview image scaling changes, I've been trying to see how the images for e.g. MIRI_4QPM images look. But at the moment there's no way (from the archive page) to get a list of programs that contain MIRI_4QPM files. Instead you need to click on various proposals and check the thumbnails for each.

### PR DESCRIPTION
**Pull request by BradleySappington**
filter by exp_type

### CODE REVIEW

**Comment by BradleySappington** [comment]
@mfixstsci @bhilbert4 - Ready for Test and review.  I had to do manual merge, so test necessary before merge.

File: jwql/website/apps/jwql/data_containers.py [hunk 1]
```diff
@@ -1509,11 +1511,13 @@ def thumbnails_ajax(inst, proposal, obs_num=None):
 
     if proposal is not None:
         dropdown_menus = {'detector': sorted(detectors),
-                          'look': THUMBNAIL_FILTER_LOOK}
+                          'look': THUMBNAIL_FILTER_LOOK,
+                          'exp_type': sorted(set(exp_types))}
```
Comment (bhilbert4) [inline jwql/website/apps/jwql/data_containers.py:1515]: Hooray for sorted!

File: jwql/website/apps/jwql/views.py [hunk 2]
```diff
@@ -226,8 +226,17 @@ def archived_proposals_ajax(request, inst):
...
     total_files = []
     proposal_viewed = []
+    proposal_exp_types = []
+    thumb_exp_types = []
+
+    # Get a set of all exposure types used in the observations associated with this proposal
+    exp_types = [exposure_type for observation in all_entries for exposure_type in observation.exptypes.split(',')]
+    exp_types = sorted(list(set(exp_types)))
+
     # The naming conventions for dropdown_menus are tightly coupled with the code, this should be changed down the line.
```
Comment (bhilbert4) [inline jwql/website/apps/jwql/views.py:239]: Could you expand on this comment in a new github issue?

File: jwql/website/apps/jwql/static/js/jwql.js [hunk 3]
```diff
@@ -778,11 +791,11 @@ function update_header_display(extension, num_extensions) {
  * @param {String} prop - Proposal ID
  * @param {List} obslist - List of observation number strings
  */
-function update_obs_options(data, inst, prop, obslist) {
+function update_obs_options(data, inst, prop, observation) {
     // Build div content
     content = 'Available observations:';
     content += '<div class="dropdown">';
-    content += '<button class="btn btn-primary dropdown-toggle" type="button" id="obs_dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Obs Nums</button>';
+    content += '<button class="btn btn-primary dropdown-toggle" type="button" id="obs_dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Obs' + observation + '</button>';
```
Comment (bhilbert4) [inline jwql/website/apps/jwql/static/js/jwql.js:798]: I like this. Making the button more informative is really helpful.

**Review by bhilbert4 — APPROVED** [review]
This looks good to me. My one general comment is that it would be nice, if the user selected a particular exp_type at the archive level, that when they clicked on a particular proposal the selection was remembered at the observation level. And for a proposal with multiple observations, that the user would be taken to the first observation with that particular exp_type. That seems like it would be some complex bookkeeping though, and can potentially be tackled in a later PR.

File: jwql/website/apps/jwql/data_containers.py (hunk 1 above)
Comment (BradleySappington) [reply jwql/website/apps/jwql/data_containers.py:1515]: A godsend

File: jwql/website/apps/jwql/views.py (hunk 2 above)
Comment (BradleySappington) [reply jwql/website/apps/jwql/views.py:239]: yup!

**Pull request merged**